### PR TITLE
Add a skill to answer confusing questions

### PR DESCRIPTION
**Pull request by ae2015**
**Describe the contribution to the taxonomy**

Given a document and a user's question that is confusing, we want LLM to respond to the question correctly without generating even more confusion. The current model tends to passively follow the lead shown by the question even if this requires twisting logic into knots and making bizarre assumptions. Common types of confusing questions include:

- Questions with a misunderstanding, such as a mixup between two concepts
- Questions based on a premise that contradicts the document or the subject matter
- Questions that do not properly match the retrieved document

The tendency of LLMs to deviate from the document, logic, and common sense when responding to confusing questions is very unhelpful to users who need to have their misunderstanding corrected rather than redoubled.

**Input given at the prompt**

Here is a rather extreme example. Merlinite 7b is provided the following context:
```
Document:

Alice has 3 green balls, Bob has 4 red balls, Charles has 5 blue balls and 1 yellow ball.
```
Then, Merlinite 7b is asked the following question:
```
Why do Alice and Bob have the same number of balls as Charles? Please explain.
```
Below are two examples of the model's response.

**Response from the original model**

Wrong response 1:
```
The number of balls that Alice and Bob have is determined by the number of colors they represent. In
this case, Alice has 3 green balls, which correspond to the three primary colors (red, blue, and yellow).
Similarly, Bob has 4 red balls, which also correspond to the four primary colors. Charles, on the other
hand, has 5 blue balls and 1 yellow ball. This means that Charles represents two primary colors instead
of one. Therefore, Alice and Bob have the same number of balls as Charles because they each represent
a different set of primary colors.
```
Wrong response 2:
```
The initial context describes that Alice has three green balls, Bob has four red balls, while Charles
has five blue balls and one yellow ball. The question then asks why Alice and Bob have the same
number of balls as Charles.

To answer this question, we first need to clarify what is meant by "the same number of balls." In this
case, it seems to refer to the total count of individual objects, not the diversity of colors or types.
Accordingly:

The response would then be:

The total number of balls for Alice and Bob combined (3 + 4) is equal to the sum of all types of balls
belonging to Charles (5+1). Therefore, in terms of quantity, they have the same amount of equipment
available between them.
```

**Response from the fine-tuned model**

(did not have an opportunity to fine tune the model)

**Contribution checklist**

- [x] The contribution was tested with `ilab generate`
- [x] No errors or warnings were produced by `ilab generate`
- [x] All [commits are signed off](https://github.com/instructlab/taxonomy/blob/main/CONTRIBUTING.md#legal) (DCO)
- [x] The `qna.yaml` file contains at least 5 `seed_examples`
- [x] The `qna.yaml` file was [linted](https://yamllint.com) and [prettified](https://onlineyamltools.com/prettify-yaml) ([yaml-validator](https://jsonformatter.org/yaml-validator) can do both)
- [x] An `attribution.txt` file in the same folder as the `qna.yaml` file.
- [x] Content does not include PII or otherwise sensitive or confidential information
- [x] Content does not include anything documented in the project's [Avoid these Topics](https://github.com/instruct-lab/community/blob/main/docs/README.md#avoid-these-topics) guidelines

### CODE REVIEW

**Comment by ae2015** [comment]
This submission contains a part of a [**short biography**](https://governors.library.ca.gov/38-Schwarzenegger.html) of a California governor from a public `library.ca.gov` website. This text lists a CA governor's full name, date of birth, place of birth, political party etc. Since PR submissions require that "content does not include PII or otherwise sensitive or confidential information", I want to double-check whether a public biography text for a California governor (or other such public figure) may constitute PII. Please advise if this example should be excluded from my submission and replaced with a different example. Thanks!

**Comment by mingxzhao** [comment]
@instructlab-bot precheck

**Comment by mingxzhao** [comment]
> This submission contains a part of a [**short biography**](https://governors.library.ca.gov/38-Schwarzenegger.html) of a California governor from a public `library.ca.gov` website. This text lists a CA governor's full name, date of birth, place of birth, political party etc. Since PR submissions require that "content does not include PII or otherwise sensitive or confidential information", I want to double-check whether a public biography text for a California governor (or other such public figure) may constitute PII. Please advise if this example should be excluded from my submission and replaced with a different example. Thanks!

Sorry for the late reply, but yes, we have decided that as long as it is a prominent public figure that has the information publicly and easily accessible, we will allow these sorts of submissions. Thank you for checking!

**Comment by jjasghar** [comment]
@instructlab-bot precheck

File: compositional_skills/writing/grounded/confusing_questions/attribution.txt
Comment (jjasghar) [inline compositional_skills/writing/grounded/confusing_questions/attribution.txt:1]: This `attribution.txt` needs to be something like: https://github.com/instructlab/taxonomy/blob/534ec6e3a18b0927aec9320bcd70a5519c43440e/knowledge/sports/overview/softball/attribution.txt not a yaml file.

**Review by jjasghar — CHANGES_REQUESTED** [review]
Please fix the `attribution.txt`

**Comment by mingxzhao** [comment]
@instructlab-bot generate

**Comment by mingxzhao** [comment]
Hey there ae2015, thank you for another though provoking PR, we had to have a discussion on this one again. Because of the fairly complex nature of the questions, there is some math and other "core skills" that seem to be very confusing to the teacher model win trying to generate synthetic data. Will close for now. Feel free to ping me if you disagree.

**Comment by ae2015** [comment]
@mingxzhao Math and "core skills" are not the critical aspect here. The critical aspect is the LLM detecting unanswerable questions and having a "brake" that alerts the model about confusion and reroutes the response from answering the question to clarifying it. One possible way to augment the synthetic data generator is to let it sometimes generate knowingly unanswerable questions paired with clarifying responses (or at least, responses telling the user that the question is unanswerable).

**Comment by ae2015** [comment]
Hi @mingxzhao and @jjasghar, before letting this pull request disappear, I wanted to ask you if it can be improved, for example, by replacing "math" with other examples. I believe that this PR addresses a very important shortcoming of the LLM, and a shortcoming that can be mitigated by training. Customers who issue inquiries to an LLM will often have incorrect assumptions due to lack of technical knowledge or because they are misled by over-simplified (too high-level) or outdated or biased information. It is critical that LLM detects confusion and corrects it rather than plays along and creates even more confusion. Maybe this particular PR is not good, but the issue it highlights is too important to just drop. The reason I think this can be mitigated by training is:
1. The LLM, upon reading its own output, should be able to detect that it makes no sense;
2. It is possible to generate synthetic training data as pairs of
- knowingly unanswerable questions and
- clarifying responses telling the user that the question is unanswerable

**Pull request closed without merging**